libp2p/pubsub/pubsub.py

```diff
@@ -320,6 +361,32 @@ def handle_subscription(self, origin_id: ID, sub_message: Any) -> None:
 
         await self.push_msg(self.host.get_id(), msg)
 
+    async def validate_msg(self, msg_forwarder: ID, msg: rpc_pb2.Message) -> bool:
+        """
+        Validate the received message
+        :param msg_forwarder: the peer who forward us the message.
+        :param msg: the message.
+        """
+        sync_topic_validators = []
+        async_topic_validator_futures = []
+        for topic_validator in self.get_msg_validators(msg):
+            if topic_validator.is_async:
+                async_topic_validator_futures.append(topic_validator.validator(msg_forwarder, msg))
+            else:
+                sync_topic_validators.append(topic_validator.validator)
+
+        for validator in sync_topic_validators:
+            if not validator(msg_forwarder, msg):
+                return False
```

Comment:
I'd recommend against a `True/False` based validation scheme in favor of an excepiton based on.  This diverges from the standard pattern we have in place for validation across most of the rest of our libraries **and** it requires return-value-checking.

If you want to stay with the `True/False` return scheme, I'd recommend a name change to something like `is_msg_valid(msg)`

Reply:
There's a similar discussion in #194 but that one is more about user facing APIs. Would also love your feedback on that.